crates/wdk-build/src/lib.rs

```diff
@@ -145,12 +145,55 @@ pub struct UmdfConfig {
     pub minimum_umdf_version_minor: Option<u8>,
 }
 
+/// Metadata providing additional context for [`std::io::Error`] failures
+///
+/// This enum provides structured information about the file system paths
+/// or operations that led to an I/O error. It can represent either single
+/// path operations or operations involving both source and destination paths.
+#[non_exhaustive]
+#[derive(Debug, Error)]
+pub enum IoErrorMetadata {
+    /// Path related to [`std::io::Error`] failure
+    #[error(r#"failed to perform an IO operation on "{path}""#)]
+    SinglePath {
```

Comment:
```suggestion
    Path {
```
Consider renaming to `Path`. `SinglePath` seems a bit awkward. That fact that the path is single is clear from the structure of the type.